### DISCUSSION TITLE
How to initialize data/collections/indexes on startup?

### DISCUSSION THREAD

**Discussion by thardy** [typegoose/mongodb-memory-server#876] [Q&A]
I want to initialize my mongodb-memory-server with the some collections, indexes, and initial data for testing. I'd like to use the same file full of shell commands I use with my Docker deployment if possible. Many of my tests involve unique indexes and testing for proper responses to duplicate inserts, etc.

I saw the answers to this question - https://github.com/nodkz/mongodb-memory-server/issues/309, but they did not mention how to programmatically execute mongo commands from a file against the newly created MongoMemoryServer.

Any help will be greatly appreciated.

**Comment by hasezoey** [comment]
> I saw the answers to this question - https://github.com/nodkz/mongodb-memory-server/issues/309, but they did not mention how to programmatically execute mongo commands from a file against the newly created MongoMemoryServer.

its the same way as you would against a normal manually run binary, the only difference is that you need to get the url / port from `instance.getUri()` as the port is random (unless set specifically).

please note that MMS *only* installs the server binary (`mongod`) **not** other binaries like `mongo` or `mongoimport` / `mongoexport` (or other binaries that may exist).

instead of `mongo` you could use the mongo nodejs driver instead.

**Comment by jeffreyflynt** [comment]
I have a `setupTests.js` file that I do this with ava. I run this before my tests run to fill the database with dummy data.

```
const setupTests = async () => {
  mongod = await MongoMemoryServer.create({
    instance: {
      port: 5001,
      dbName: "docs",
      ip: "0.0.0.0",
    },
    binary: {
      version: "6.0.14",
      checkMD5: true,
    },
  });

  const client = new MongoClient(process.env.MONGOURL, { family: 4 });

  await client.connect();

  const db = client.db("docs");
  await db.collection("supportItems").insertOne(helpTopic1);
  await db.collection("users").insertMany([user1, user2, user3, user4]);
  await db.collection("teams").insertMany([team1, team2, team3, team4]);
  ```
